File: backend/routes/crimeTypes.js
import { Router } from "express";
import { pool } from "../db.js";

const router = Router();

// GET /api/crime-types — replaces the mock's hardcoded `presetTags`
router.get("/", async (req, res) => {
  try {
    const { rows } = await pool.query("SELECT crime_name FROM crime_types ORDER BY crime_name");
    res.json(rows.map((r) => r.crime_name));
  } catch (err) {
    console.error("GET /api/crime-types failed", err);
    res.status(500).json({ error: "Failed to load crime types" });
  }
});

export default router;

File: backend/routes/crime_types.py
from fastapi import APIRouter, Depends, Request

from ..errors import api_errors
from ..security import require_auth

router = APIRouter(
    prefix="/api/crime-types", tags=["Crime types"], dependencies=[Depends(require_auth)]
)


# Accept both trailing-slash forms while documenting one endpoint.
@router.get("/", include_in_schema=False)
@router.get("")
async def get_crime_types(request: Request):
    with api_errors("Failed to load crime types"):
        rows = await request.app.state.db.query(
            "SELECT crime_name FROM crime_types ORDER BY crime_name"
        )
        return [row["crime_name"] for row in rows]
